google/cloud/speech/CMakeLists.txt

```diff
@@ -50,10 +50,24 @@ google_cloud_cpp_grpcpp_library(
 external_googleapis_set_version_and_alias(speech_protos)
 target_link_libraries(google_cloud_cpp_speech_protos PUBLIC ${proto_deps})
 
+unset(mocks_globs)
+unset(source_globs)
+set(service_dirs ";v2/")
```

Comment:
if you are planning to use something like a list, then do so:

```
set(service_dirs "" "v2/")
set(internal_dirs "" "internal/")
set(extensions "*.h" ".cc")
```

But all of this seems overly complicated, I think something like

```
file(GLOB source_files RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
       "*.h" "*.cc" "internal/*.h" "intermal/*.cc"
       "v2/*.h" "v2/*.cc" "v2/internal/*.h" "v2/intermal/*.cc")
```

Is easier to read.

Reply:
Ack, I unrolled the `internal_dirs` and `extensions`, but I left the `service_dirs` loop. I think it offers a general solution that we can use in other libraries. e.g. bigquery might (eventually) look like:

```
set(service_dirs "analyticshub/v1/" "connection/v1/"
                 "datatransfer/v1/" "migration/v2/"
                 "reservation/v1/" "storage/v1/" "v2/")
```